template/src/components/views/DatasetList.js

```diff
@@ -0,0 +1,161 @@
+import React, { useState } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+
+import { addLayer, addSource, removeLayer, removeSource } from 'config/cartoSlice';
+import { selectOAuthCredentials } from 'config/oauthSlice';
+
+import { makeStyles } from '@material-ui/core/styles';
+import {
+  Button,
+  Dialog,
+  DialogTitle,
+  DialogContent,
+  DialogContentText,
+  DialogActions,
+  IconButton,
+  List,
+  ListItem,
+  ListItemSecondaryAction,
+  ListItemText,
+  Typography,
+} from '@material-ui/core';
+
+import { ChevronRight, HighlightOff } from '@material-ui/icons';
+
+const useStyles = makeStyles((theme) => ({
+  dataset: {
+    maxWidth: '100%',
+  },
+  datasetName: {
+    overflow: 'hidden',
+    textOverflow: 'ellipsis',
+  },
+}));
+
+function DatasetList(props) {
+  const credentials = useSelector(selectOAuthCredentials);
+  const { oauthLayer } = useSelector((state) => state.carto.layers);
+
+  const dispatch = useDispatch();
+  const [open, setOpen] = useState(false);
+  const [selectedDataset, setSelectedDataset] = useState({});
+  const classes = useStyles();
+
+  const handleClickOpen = (dataset) => {
+    setSelectedDataset(dataset);
+    setOpen(true);
+  };
+  const handleClose = () => {
+    setOpen(false);
+  };
+
+  // Load dataset & layer to store (so to Map)
+  const loadDataset = () => {
+    handleClose();
+
+    const { name: datasetName, table_schema: schema } = selectedDataset;
+    const dataSourceCredentials = { ...credentials, username: schema };
+
+    dispatch(
+      addSource({
+        id: 'oauthSource',
+        data: `SELECT * FROM "${schema}".${datasetName}`,
+        credentials: dataSourceCredentials,
+      })
+    );
+
+    dispatch(addLayer({ id: 'oauthLayer', source: 'oauthSource', name: datasetName }));
+  };
+
+  // Remove dataset & layer from store (so from Map)
+  const removeDataset = () => {
+    dispatch(removeSource('oauthSource'));
+    dispatch(removeLayer('oauthLayer'));
+  };
+
+  if (props.datasets.length === 0) {
```

Comment:
When login is running this component will show `No datasets available` we should consider adding a Loading component. Ping design for this. They talked yesterday about a default spinner.